File: InternMobileApp/src/screens/tabs/ProfileScreen.js
import React, { useState } from 'react';
import { View, Text, StyleSheet, ScrollView, TouchableOpacity, TextInput, ActivityIndicator, Alert, Image } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useAuth } from '../../context/AuthContext';
import { useDashboard } from '../../context/DashboardContext';
import { supabase } from '../../lib/supabase';
import { colors } from '../../theme/colors';

export default function ProfileScreen() {
    const { profile } = useAuth();
    const { currentTeam } = useDashboard();
    const currentUser = profile?.userProfile;

    const [linkedIn, setLinkedIn] = useState(currentUser?.social_links?.linkedin || '');
    const [bio, setBio] = useState(currentUser?.bio || '');
    const [saving, setSaving] = useState(false);

    const name = currentUser?.full_name || currentUser?.email?.split('@')[0] || 'Intern';
    const initial = name[0]?.toUpperCase() || 'I';
    const isPaid = currentUser?.status === 'active';
    const streak = currentUser?.current_streak || 0;
    
    const calculateLevel = (s) => {
        if (s >= 30) return 'Expert';
        if (s >= 14) return 'Advanced';
        if (s >= 7) return 'Intermediate';
        return 'Beginner';
    };
    
    const level = calculateLevel(streak);
    const title = currentUser?.title || 'Intern';
    const phone = currentUser?.phone || 'Not set';
    const batchName = currentTeam?.batches?.name || 'Not assigned';

    const handleLogout = async () => {
        await supabase.auth.signOut();
    };

    const handleSave = async () => {
        setSaving(true);
        try {
            const updates = {
                bio,
                social_links: { ...currentUser?.social_links, linkedin: linkedIn }
            };

            const { error } = await supabase
                .from('profiles')
                .update(updates)
                .eq('id', currentUser.id);

            if (error) throw error;
            Alert.alert('Success', 'Profile updated successfully.');
        } catch (err) {
            console.error('Error saving profile:', err);
            Alert.alert('Error', 'Could not update profile.');
        } finally {
            setSaving(false);
        }
    };

    return (
        <View style={styles.container}>
            <View style={styles.header}>
                <Text style={styles.headerTitle}>Profile</Text>
            </View>

            <ScrollView style={styles.body} contentContainerStyle={{ padding: 20 }}>
                {/* Profile Header */}
                <View style={styles.profileHeader}>
                    <View style={styles.avatarContainer}>
                        {currentUser?.avatar_url ? (
                            <Image source={{ uri: currentUser.avatar_url }} style={styles.avatarImage} />
                        ) : (
                            <Text style={styles.avatarText}>{initial}</Text>
                        )}
                    </View>
                    <Text style={styles.nameText}>{name}</Text>
                    <Text style={styles.titleText}>{title}</Text>
                    <View style={[styles.statusBadge, { backgroundColor: isPaid ? '#d1fae5' : '#fef3c7' }]}>
                        <Ionicons name={isPaid ? "checkmark-circle" : "time"} size={14} color={isPaid ? '#065f46' : '#92400e'} style={{ marginRight: 4 }} />
                        <Text style={[styles.statusBadgeText, { color: isPaid ? '#065f46' : '#92400e' }]}>{isPaid ? 'Active Intern' : 'Pending Payment'}</Text>
                    </View>
                </View>

                {/* Stats */}
                <View style={styles.statsGrid}>
                    <View style={styles.statCard}>
                        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
                            <Ionicons name="flame" size={24} color="#f97316" style={{ marginRight: 6 }} />
                            <Text style={[styles.statValue, { color: '#f97316' }]}>{streak}</Text>
                        </View>
                        <Text style={styles.statLabel}>Day Streak</Text>
                    </View>
                    <View style={styles.statCard}>
                        <Text style={[styles.statValue, { color: '#8b5cf6' }]}>{level}</Text>
                        <Text style={styles.statLabel}>Level</Text>
                    </View>
                    <View style={styles.statCard}>
                        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
                            <Ionicons name="layers" size={20} color="#3b82f6" style={{ marginRight: 6 }} />
                            <Text style={[styles.statValue, { color: '#3b82f6', fontSize: 18 }]}>
                                {batchName === 'Not assigned' ? 'None' : batchName.replace('BATCH ', '')}
                            </Text>
                        </View>
                        <Text style={styles.statLabel}>Batch</Text>
                    </View>
                    <View style={styles.statCard}>
                        <Ionicons name={isPaid ? "checkmark-circle" : "alert-circle"} size={28} color={isPaid ? '#10b981' : '#f59e0b'} />
                        <Text style={styles.statLabel}>{isPaid ? 'Paid' : 'Pending'}</Text>
                    </View>
                </View>

                {/* Edit Form */}
                <View style={styles.card}>
                    <View style={{ flexDirection: 'row', alignItems: 'center', marginBottom: 15 }}>
                        <Ionicons name="person-circle" size={20} color={colors.primary} style={{ marginRight: 8 }} />
                        <Text style={styles.sectionTitle}>Personal Information</Text>
                    </View>

                    <Text style={styles.label}>Full Name</Text>
                    <TextInput style={[styles.input, styles.disabledInput]} value={name} editable={false} />

                    <Text style={styles.label}>Email</Text>
                    <TextInput style={[styles.input, styles.disabledInput]} value={currentUser?.email} editable={false} />

                    <Text style={styles.label}>Phone Number</Text>
                    <TextInput style={[styles.input, styles.disabledInput]} value={phone} editable={false} />

                    <Text style={styles.label}>LinkedIn Profile</Text>
                    <TextInput 
                        style={styles.input} 
                        value={linkedIn} 
                        onChangeText={setLinkedIn} 
                        placeholder="https://linkedin.com/in/..." 
                        placeholderTextColor="#475569"
                    />

                    <Text style={styles.label}>Bio</Text>
                    <TextInput 
                        style={[styles.input, { height: 80, textAlignVertical: 'top' }]} 
                        value={bio} 
                        onChangeText={setBio} 
                        placeholder="Tell us about yourself..." 
                        placeholderTextColor="#475569"
                        multiline
                    />

                    <TouchableOpacity style={styles.saveButton} onPress={handleSave} disabled={saving}>
                        {saving ? (
                            <ActivityIndicator size="small" color={colors.background} />
                        ) : (
                            <>
                                <Ionicons name="save" size={18} color={colors.background} style={{ marginRight: 8 }} />
                                <Text style={styles.saveButtonText}>Save Changes</Text>
                            </>
                        )}
                    </TouchableOpacity>
                </View>

                {/* Logout */}
                <TouchableOpacity style={styles.logoutButton} onPress={handleLogout}>
                    <Ionicons name="log-out-outline" size={20} color="#ef4444" style={{ marginRight: 8 }} />
                    <Text style={styles.logoutText}>Log Out</Text>
                </TouchableOpacity>

            </ScrollView>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: colors.background,
    },
    header: {
        paddingTop: 60,
        paddingHorizontal: 20,
        paddingBottom: 20,
        backgroundColor: colors.surface,
        borderBottomWidth: 1,
        borderBottomColor: '#1f1f2e',
    },
    headerTitle: {
        fontSize: 28,
        fontWeight: 'bold',
        color: colors.text,
    },
    body: {
        flex: 1,
    },
    profileHeader: {
        alignItems: 'center',
        marginBottom: 25,
    },
    avatarContainer: {
        width: 100,
        height: 100,
        borderRadius: 50,
        backgroundColor: 'rgba(34, 216, 122, 0.2)',
        alignItems: 'center',
        justifyContent: 'center',
        marginBottom: 15,
        borderWidth: 2,
        borderColor: colors.primary,
        overflow: 'hidden',
    },
    avatarImage: {
        width: '100%',
        height: '100%',
    },
    avatarText: {
        fontSize: 40,
        fontWeight: 'bold',
        color: colors.primary,
    },
    nameText: {
        fontSize: 22,
        fontWeight: 'bold',
        color: colors.text,
        marginBottom: 4,
    },
    titleText: {
        fontSize: 16,
        color: '#94a3b8',
        marginBottom: 12,
    },
    statusBadge: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingHorizontal: 12,
        paddingVertical: 6,
        borderRadius: 20,
    },
    statusBadgeText: {
        fontSize: 13,
        fontWeight: '600',
    },
    statsGrid: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        justifyContent: 'space-between',
        marginBottom: 20,
    },
    statCard: {
        width: '48%',
        backgroundColor: colors.surface,
        borderRadius: 16,
        padding: 15,
        alignItems: 'center',
        justifyContent: 'center',
        marginBottom: 15,
        borderWidth: 1,
        borderColor: '#1f1f2e',
    },
    statValue: {
        fontSize: 22,
        fontWeight: 'bold',
        marginBottom: 4,
    },
    statLabel: {
        fontSize: 13,
        color: '#94a3b8',
    },
    card: {
        backgroundColor: colors.surface,
        borderRadius: 16,
        padding: 20,
        marginBottom: 20,
        borderWidth: 1,
        borderColor: '#1f1f2e',
    },
    sectionTitle: {
        fontSize: 18,
        fontWeight: 'bold',
        color: colors.text,
    },
    label: {
        fontSize: 13,
        color: '#94a3b8',
        marginBottom: 8,
        marginTop: 15,
    },
    input: {
        backgroundColor: 'rgba(255,255,255,0.05)',
        borderWidth: 1,
        borderColor: 'rgba(255,255,255,0.1)',
        borderRadius: 8,
        padding: 12,
        color: colors.text,
        fontSize: 15,
    },
    disabledInput: {
        opacity: 0.6,
        backgroundColor: 'rgba(255,255,255,0.02)',
    },
    saveButton: {
        backgroundColor: colors.primary,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        paddingVertical: 14,
        borderRadius: 10,
        marginTop: 25,
    },
    saveButtonText: {
        color: colors.background,
        fontWeight: 'bold',
        fontSize: 16,
    },
    logoutButton: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        paddingVertical: 14,
        borderRadius: 10,
        backgroundColor: 'rgba(239, 68, 68, 0.1)',
        borderWidth: 1,
        borderColor: 'rgba(239, 68, 68, 0.2)',
        marginBottom: 40,
    },
    logoutText: {
        color: '#ef4444',
        fontWeight: 'bold',
        fontSize: 16,
    },
});

File: intern_mobile_app_flutter/lib/screens/tabs/profile_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../../core/supabase_client.dart';
import '../../providers/auth_provider.dart';
import '../../providers/dashboard_provider.dart';
import '../../theme/colors.dart';

class ProfileScreen extends StatefulWidget {
  const ProfileScreen({super.key});

  @override
  State<ProfileScreen> createState() => _ProfileScreenState();
}

class _ProfileScreenState extends State<ProfileScreen> {
  final TextEditingController _linkedInCtrl = TextEditingController();
  final TextEditingController _bioCtrl = TextEditingController();
  bool _saving = false;
  bool _initialized = false;

  @override
  void didChangeDependencies() {
    super.didChangeDependencies();
    if (!_initialized) {
      final auth = Provider.of<AuthProvider>(context, listen: false);
      final currentUser = auth.profile?['userProfile'];
      if (currentUser != null) {
        _linkedInCtrl.text = currentUser['social_links']?['linkedin'] ?? '';
        _bioCtrl.text = currentUser['bio'] ?? '';
      }
      _initialized = true;
    }
  }

  @override
  void dispose() {
    _linkedInCtrl.dispose();
    _bioCtrl.dispose();
    super.dispose();
  }

  String _calculateLevel(int s) {
    if (s >= 30) return 'Expert';
    if (s >= 14) return 'Advanced';
    if (s >= 7) return 'Intermediate';
    return 'Beginner';
  }

  Future<void> _handleLogout() async {
    await SupabaseClientConfig.client.auth.signOut();
  }

  Future<void> _handleSave() async {
    final auth = Provider.of<AuthProvider>(context, listen: false);
    final currentUser = auth.profile?['userProfile'];
    if (currentUser == null) return;

    setState(() => _saving = true);
    try {
      final currentSocialLinks = currentUser['social_links'] ?? {};
      final newSocialLinks = Map<String, dynamic>.from(currentSocialLinks);
      newSocialLinks['linkedin'] = _linkedInCtrl.text;

      final updates = {
        'bio': _bioCtrl.text,
        'social_links': newSocialLinks,
      };

      await SupabaseClientConfig.client
          .from('profiles')
          .update(updates)
          .eq('id', currentUser['id']);

      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(const SnackBar(content: Text('Profile updated successfully.')));
        auth.refreshProfile(); // refresh
      }
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(const SnackBar(content: Text('Could not update profile.')));
      }
    } finally {
      if (mounted) setState(() => _saving = false);
    }
  }

  Widget _buildStatCard(Widget top, String label) {
    return Container(
      width: (MediaQuery.of(context).size.width - 40 - 15) / 2,
      margin: const EdgeInsets.only(bottom: 15),
      padding: const EdgeInsets.all(15),
      decoration: BoxDecoration(
        color: AppColors.surface,
        borderRadius: BorderRadius.circular(16),
        border: Border.all(color: const Color(0xFF1f1f2e)),
      ),
      child: Column(
        children: [
          top,
          const SizedBox(height: 4),
          Text(label, style: const TextStyle(color: Color(0xFF94a3b8), fontSize: 13)),
        ],
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    final auth = Provider.of<AuthProvider>(context);
    final dashboard = Provider.of<DashboardProvider>(context);

    final currentUser = auth.profile?['userProfile'];
    
    final fullName = currentUser?['full_name'] ?? currentUser?['email']?.split('@')[0] ?? 'Intern';
    final initial = fullName.isNotEmpty ? fullName[0].toUpperCase() : 'I';
    final isPaid = currentUser?['status'] == 'active';
    final streak = currentUser?['current_streak'] ?? 0;
    final level = _calculateLevel(streak);
    final title = currentUser?['title'] ?? 'Intern';
    final phone = currentUser?['phone'] ?? 'Not set';
    final batchName = dashboard.currentTeam?['batches']?['name'] ?? 'Not assigned';
    final email = currentUser?['email'] ?? '';
    final avatarUrl = currentUser?['avatar_url'];

    return Scaffold(
      backgroundColor: AppColors.background,
      body: SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Container(
              padding: const EdgeInsets.only(top: 20, left: 20, right: 20, bottom: 20),
              width: double.infinity,
              decoration: const BoxDecoration(
                color: AppColors.surface,
                border: Border(bottom: BorderSide(color: Color(0xFF1f1f2e))),
              ),
              child: const Text('Profile', style: TextStyle(color: AppColors.text, fontSize: 28, fontWeight: FontWeight.bold)),
            ),
            Expanded(
              child: RefreshIndicator(
                color: AppColors.primary,
                backgroundColor: AppColors.card,
                onRefresh: () async {
                  final authProvider = Provider.of<AuthProvider>(context, listen: false);
                  final dashProvider = Provider.of<DashboardProvider>(context, listen: false);
                  await authProvider.refreshProfile();
                  final userId = authProvider.profile?['userProfile']?['id'];
                  if (userId != null) {
                    await dashProvider.fetchDashboardData(userId, forceRefresh: true);
                  }
                },
                child: ListView(
                  padding: const EdgeInsets.all(20),
                  children: [
                  // Profile Header
                  Column(
                    children: [
                      Container(
                        width: 100, height: 100,
                        margin: const EdgeInsets.only(bottom: 15),
                        decoration: BoxDecoration(
                          color: const Color(0x3322d87a),
                          shape: BoxShape.circle,
                          border: Border.all(color: AppColors.primary, width: 2),
                        ),
                        alignment: Alignment.center,
                        clipBehavior: Clip.hardEdge,
                        child: avatarUrl != null
                            ? Image.network(avatarUrl, width: 100, height: 100, fit: BoxFit.cover)
                            : Text(initial, style: const TextStyle(color: AppColors.primary, fontSize: 40, fontWeight: FontWeight.bold)),
                      ),
                      Text(fullName, style: const TextStyle(color: AppColors.text, fontSize: 22, fontWeight: FontWeight.bold)),
                      const SizedBox(height: 4),
                      Text(title, style: const TextStyle(color: Color(0xFF94a3b8), fontSize: 16)),
                      const SizedBox(height: 12),
                      Container(
                        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                        decoration: BoxDecoration(
                          color: isPaid ? const Color(0xFFd1fae5) : const Color(0xFFfef3c7),
                          borderRadius: BorderRadius.circular(20),
                        ),
                        child: Row(
                          mainAxisSize: MainAxisSize.min,
                          children: [
                            Icon(isPaid ? Icons.check_circle : Icons.access_time, size: 14, color: isPaid ? const Color(0xFF065f46) : const Color(0xFF92400e)),
                            const SizedBox(width: 4),
                            Text(isPaid ? 'Active Intern' : 'Pending Payment', style: TextStyle(color: isPaid ? const Color(0xFF065f46) : const Color(0xFF92400e), fontSize: 13, fontWeight: FontWeight.bold)),
                          ],
                        ),
                      )
                    ],
                  ),
                  const SizedBox(height: 25),

                  // Stats Grid
                  Wrap(
                    spacing: 15,
                    children: [
                      _buildStatCard(
                        Row(mainAxisSize: MainAxisSize.min, children: [
                          const Icon(Icons.local_fire_department, size: 24, color: Color(0xFFf97316)),
                          const SizedBox(width: 6),
                          Text(streak.toString(), style: const TextStyle(color: Color(0xFFf97316), fontSize: 22, fontWeight: FontWeight.bold)),
                        ]),
                        'Day Streak',
                      ),
                      _buildStatCard(
                        Text(level, style: const TextStyle(color: Color(0xFF8b5cf6), fontSize: 22, fontWeight: FontWeight.bold)),
                        'Level',
                      ),
                      _buildStatCard(
                        Row(mainAxisSize: MainAxisSize.min, children: [
                          const Icon(Icons.layers, size: 20, color: Color(0xFF3b82f6)),
                          const SizedBox(width: 6),
                          Text(batchName == 'Not assigned' ? 'None' : batchName.replaceAll('BATCH ', ''), style: const TextStyle(color: Color(0xFF3b82f6), fontSize: 18, fontWeight: FontWeight.bold)),
                        ]),
                        'Batch',
                      ),
                      _buildStatCard(
                        Icon(isPaid ? Icons.check_circle : Icons.error, size: 28, color: isPaid ? const Color(0xFF10b981) : const Color(0xFFf59e0b)),
                        isPaid ? 'Paid' : 'Pending',
                      ),
                    ],
                  ),
                  const SizedBox(height: 5),

                  // Edit Form
                  Container(
                    padding: const EdgeInsets.all(20),
                    decoration: BoxDecoration(
                      color: AppColors.surface,
                      borderRadius: BorderRadius.circular(16),
                      border: Border.all(color: const Color(0xFF1f1f2e)),
                    ),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          children: const [
                            Icon(Icons.person, size: 20, color: AppColors.primary),
                            SizedBox(width: 8),
                            Text('Personal Information', style: TextStyle(color: AppColors.text, fontSize: 18, fontWeight: FontWeight.bold)),
                          ],
                        ),
                        const SizedBox(height: 15),
                        
                        const Text('Full Name', style: TextStyle(color: Color(0xFF94a3b8), fontSize: 13)),
                        const SizedBox(height: 8),
                        _buildDisabledInput(fullName),
                        const SizedBox(height: 15),

                        const Text('Email', style: TextStyle(color: Color(0xFF94a3b8), fontSize: 13)),
                        const SizedBox(height: 8),
                        _buildDisabledInput(email),
                        const SizedBox(height: 15),

                        const Text('Phone Number', style: TextStyle(color: Color(0xFF94a3b8), fontSize: 13)),
                        const SizedBox(height: 8),
                        _buildDisabledInput(phone),
                        const SizedBox(height: 15),

                        const Text('LinkedIn Profile', style: TextStyle(color: Color(0xFF94a3b8), fontSize: 13)),
                        const SizedBox(height: 8),
                        TextField(
                          controller: _linkedInCtrl,
                          style: const TextStyle(color: AppColors.text, fontSize: 15),
                          decoration: _inputDecoration('https://linkedin.com/in/...'),
                        ),
                        const SizedBox(height: 15),

                        const Text('Bio', style: TextStyle(color: Color(0xFF94a3b8), fontSize: 13)),
                        const SizedBox(height: 8),
                        TextField(
                          controller: _bioCtrl,
                          maxLines: 4,
                          style: const TextStyle(color: AppColors.text, fontSize: 15),
                          decoration: _inputDecoration('Tell us about yourself...'),
                        ),
                        const SizedBox(height: 25),

                        ElevatedButton.icon(
                          style: ElevatedButton.styleFrom(
                            backgroundColor: AppColors.primary,
                            minimumSize: const Size(double.infinity, 48),
                            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                          ),
                          onPressed: _saving ? null : _handleSave,
                          icon: _saving ? const SizedBox() : const Icon(Icons.save, color: AppColors.background, size: 18),
                          label: _saving 
                              ? const SizedBox(width: 20, height: 20, child: CircularProgressIndicator(color: AppColors.background, strokeWidth: 2))
                              : const Text('Save Changes', style: TextStyle(color: AppColors.background, fontSize: 16, fontWeight: FontWeight.bold)),
                        )
                      ],
                    ),
                  ),
                  const SizedBox(height: 20),

                  // Logout
                  OutlinedButton.icon(
                    style: OutlinedButton.styleFrom(
                      backgroundColor: const Color(0x1Aef4444),
                      side: const BorderSide(color: Color(0x33ef4444)),
                      minimumSize: const Size(double.infinity, 48),
                      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                    ),
                    onPressed: _handleLogout,
                    icon: const Icon(Icons.logout, color: Color(0xFFef4444), size: 20),
                    label: const Text('Log Out', style: TextStyle(color: Color(0xFFef4444), fontSize: 16, fontWeight: FontWeight.bold)),
                  ),
                  const SizedBox(height: 40),
                ],
              ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildDisabledInput(String value) {
    return Container(
      width: double.infinity,
      padding: const EdgeInsets.all(12),
      decoration: BoxDecoration(
        color: const Color(0x05FFFFFF),
        border: Border.all(color: const Color(0x1AFFFFFF)),
        borderRadius: BorderRadius.circular(8),
      ),
      child: Text(value, style: TextStyle(color: AppColors.text.withValues(alpha: 0.6), fontSize: 15)),
    );
  }

  InputDecoration _inputDecoration(String hint) {
    return InputDecoration(
      hintText: hint,
      hintStyle: const TextStyle(color: Color(0xFF475569)),
      filled: true,
      fillColor: const Color(0x0DFFFFFF),
      enabledBorder: OutlineInputBorder(borderRadius: BorderRadius.circular(8), borderSide: const BorderSide(color: Color(0x1AFFFFFF))),
      focusedBorder: OutlineInputBorder(borderRadius: BorderRadius.circular(8), borderSide: const BorderSide(color: AppColors.primary)),
      contentPadding: const EdgeInsets.symmetric(horizontal: 12, vertical: 12),
    );
  }
}
